### PR TITLE
Bicaws 3043 switching survey

### PR DESCRIPTION
**Pull request by Senny10**

### CODE REVIEW

File: cypress/e2e/switchingFeedback.cy.ts [hunk 1]
```diff
@@ -0,0 +1,128 @@
...
+    cy.get("h3").contains("Why have you decided to switch version of Bichard?")
+    cy.get("h5").contains("Select one of the below option")
+    cy.get("label").contains(
+      "I have found an issue(s) when using the new version of Bichard which is blocking me from completing my task."
+    )
+    cy.get("label").contains("I prefer working in the old version, and I dislike working in the new version.")
+    cy.get("label").contains("Other (please specify)")
+  })
+
+  TestCases.forEach(([testName, testInput]) => {
```
Comment (C-gyorfi) [inline cypress/e2e/switchingFeedback.cy.ts:63]: Wow, this test matrix is complex, but I guess makes sense and reduces a lot of code, the test data is fairly readable!

File: src/utils/shouldShowSwitchingFeedbackForm.ts [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+    date.getUTCMonth(),
+    date.getUTCDate(),
+    date.getUTCHours(),
+    date.getUTCMinutes(),
+    date.getUTCSeconds(),
+    date.getUTCMilliseconds()
+  )
+  const hoursBehindInMs = addHours(utcDate, -SWITCHING_FEEDBACK_FORM_FREQUENCY_IN_HOURS).getTime()
+
+  console.log(lastFeedbackFormSubmission)
```
Comment (C-gyorfi) [inline src/utils/shouldShowSwitchingFeedbackForm.ts:17]: Assuming we don't need this console log

File: test/services/getLastSwitchingFormSubmission.integration.test.ts [hunk 3]
```diff
@@ -0,0 +1,109 @@
...
+import User from "services/entities/User"
+import getUser from "services/getUser"
+import insertSurveyFeedback from "services/insertSurveyFeedback"
+import { DataSource, Repository } from "typeorm"
+import { isError } from "types/Result"
+import { SurveyFeedbackType, SwitchingFeedbackResponse } from "types/SurveyFeedback"
+import getDataSource from "../../src/services/getDataSource"
+import getLastSwitchingFormSubmission from "../../src/services/getLastSwitchingFormSubmission"
+import deleteFromEntity from "../utils/deleteFromEntity"
+describe("getLastSwitchingFormSubmission", () => {
```
Comment (C-gyorfi) [inline test/services/getLastSwitchingFormSubmission.integration.test.ts:11]: Nit picking, could add an empty line here :D

**Review by C-gyorfi — APPROVED** [review]
Nice work! 🎉

File: .circleci/config.yml [hunk 4]
```diff
@@ -235,11 +235,11 @@ jobs:
       WORKSPACE: local-next
       AUTH_TYPE: user-service
     steps:
-      - setup
-      - run: ./.circleci/scripts/build-ui.sh
       - run:
           name: Clone bichard7-next-tests
           command: git clone --depth 1 https://github.com/ministryofjustice/bichard7-next-tests.git ~/bichard7-next-tests
+      - setup
```
[the remark this answers is not in the document]
Comment (emadkaramad) [reply .circleci/config.yml:241]: During the setup step, node_modules is restored in ~/bichard7-next-tests and Git clone command fails if the directory already exists. So we need to clone the repo first and then restore the cache.

**Review by C-gyorfi — APPROVED** [review]

File: src/pages/switching-feedback.tsx [hunk 5]
```diff
@@ -0,0 +1,344 @@
...
+  }, [])
+
+  const getRemainingLength = useCallback(
+    () => MAX_FEEDBACK_LENGTH - (formState.feedback?.length || 0),
+    [formState.feedback]
+  )
+
+  return (
+    <Layout user={user}>
+      <Heading as="h2" size="LARGE" aria-label="Switching Feedback">
```
Comment (richrace) [inline src/pages/switching-feedback.tsx:163]: This needs to be `<Head>`

**Pull request merged**